app/services/activities/import_from_csv.rb

```diff
@@ -21,31 +21,44 @@ def import(activities)
     end
 
     def import_row(row, index)
-      activity = Activity.by_roda_identifier(row["RODA ID"])
-
-      if activity.nil?
-        add_error(index, :roda_id, row["RODA ID"], I18n.t("importer.errors.activity.not_found"))
+      if row["RODA ID"].blank? && row["RODA ID Fragment"].present? && row["Parent RODA ID"].present?
+        action = create_activity(row, index)
+      elsif row["RODA ID"].present?
+        action = update_activity(row, index)
       else
-        updater = ActivityUpdater.new(activity, @organisation, row)
-        updater.update
+        add_error(index, :roda_id, row["RODA ID"], I18n.t("importer.errors.activity.cannot_create_or_update")) && return
+      end
```

Comment:
It might be worth tightening the ambiguity detection here. This will catch the case where `RODA ID` is blank and either `RODA ID Fragment` or `Parent RODA ID` is blank, but it won't catch some other cases, for example both `RODA ID` and `RODA ID Fragment` are present. I'd suggest:

- create an activity if `RODA ID` is blank, `RODA ID Fragment` is present, and `Parent RODA ID` is present
- update an activity if `RODA ID` is present, `RODA ID Fragment` is blank, and `Parent RODA ID` is blank (i.e. exact inverse of previous case)
- raise an ambiguity error in all other cases

Alternatively I'd select create vs update based purely on the presence of `RODA ID` and treat the absence or presence of the other fields as errors, but this might produce less informative errors than the above.

Reply:
In fact in the case where `RODA ID` and `RODA ID Fragment` are both present, this could update an existing activity and overwrite its `roda_identifier_fragment` field, so I'd treat this as a logic bug rather than just unclear error messaging.

Reply:
@jcoglan I've added another commit with some explanations to try and tighten up this ambiguity detection 👍

Reply:
@pezholio That looks correct to me and I believe all cases have test coverage. Looks good 👍